src/search/node/FindInFilesDomain.js

```diff
@@ -0,0 +1,661 @@
+        // this always returns a regexp even for simple strings. Reconcile.
+        if (!queryInfo || !queryInfo.query) {
+            return {empty: true};
+        }
+
+        // For now, treat all matches as multiline (i.e. ^/$ match on every line, not the whole
+        // document). This is consistent with how single-file find works. Eventually we should add
+        // an option for this.
+        var flags = "gm";
+        if (!queryInfo.isCaseSensitive) {
+            flags += "i";
+        }
+
+        // Is it a (non-blank) regex?
+        if (queryInfo.isRegexp) {
+            try {
+                queryExpr = new RegExp(queryInfo.query, flags);
+            } catch (e) {
+                return {valid: false, error: e.message};
+            }
+        } else {
+            // Query is a plain string. Turn it into a regexp
+            queryExpr = new RegExp(regexEscape(queryInfo.query), flags);
+        }
+        return {valid: true, queryExpr: queryExpr};
+    }
+    
+    /**
+     * Crawls through the files in the project ans stores them in cache. Since that could take a while
+     * we do it in batches so that node wont be blocked.
+     */
+    function fileCrawler() {
+        if (!files || (files && files.length === 0)) {
+            setTimeout(fileCrawler, 1000);
+            return;
+        }
+        var i = 0;
+        for (i = 0; i < 10 && currentCrawlIndex < files.length; i = i + 1) {
+            getFileContentsForFile(files[currentCrawlIndex]);
+            currentCrawlIndex++;
+        }
+        if (currentCrawlIndex < files.length) {
+            console.log("crawling scheduled");
+            crawlComplete = false;
+            setImmediate(fileCrawler);
+        } else {
+            crawlComplete = true;
+            if (!crawlEventSent) {
+                crawlEventSent = true;
+                _domainManager.emitEvent("FindInFiles", "crawlComplete");
+            }
+            setTimeout(fileCrawler, 1000);
+        }
+    }
+
+    /**
+     * Init for project, resets the old project cache, and sets the crawler function to
+     * restart the file crawl
+     * @param   {array} fileList an array of files
+     */
+    function initCache(fileList) {
+        console.log("cache change");
+        files = fileList;
+        currentCrawlIndex = 0;
+        clearProjectCache();
+        crawlEventSent = false;
+    }
+    
+    /**
+     * Counts the number of matches matching the queryExpr in the given contents
+     * @param   {String} contents  The contents to search on
+     * @param   {Object} queryExpr
+     * @returns {number} number of matches
+     */
+    function countNumMatches(contents, queryExpr) {
+        if (!contents) {
+            return 0;
+        }
+        var matches = contents.match(queryExpr);
+        if (matches && isNaN(matches.length)) {
+            console.log('contents for nan' + contents);
+            console.log(JSON.stringify(matches));
```

Comment:
matches would be either `null' (if no matches) or array. So isNaN check is not required as it will always be true if matches is not null. You can remove this block.